frontend/ui/data/css_tweaks.lua

```diff
@@ -326,54 +326,115 @@ This is just an example, that will need to be adapted into a user style tweak.]]
-            description = _([[Decrease font size of footnotes text displayed in pages.]]),
-            -- Include the whole content of previous style, so toggling this one is enough
-            css = [[
-/* EPUB attributes: */
-*[type~="note"], *[type~="footnote"],
+                ]],
+            },
+            {
+                id = "footnote-inpage_epub_smaller";
+                title = _("In-page EPUB footnotes (smaller)"),
+                description = _([[
+Show EPUB footnotes text at the bottom of pages that contain links to them.
+This only works with footnotes that have specific attributes set by the publisher.]]),
+                css = [[
+*[type~="note"],
+*[type~="footnote"],
 *[type~="rearnote"],
-*[role~="doc-note"], *[role~="doc-footnote"],
-*[role~="doc-rearnote"],
-/* Wikipedia EPUBs footnotes: */
-ol.references > li,
-/* Classic footnotes class names: */
-.footnote, .note, .note1
+*[role~="doc-note"],
+*[role~="doc-footnote"],
+*[role~="doc-rearnote"]
 {
     -cr-hint: footnote-inpage;
+    margin-top: 0 !important;
+    margin-bottom: 0 !important;
     font-size: 80% !important;
+}
+                ]],
+                separator = true,
+            },
+            {
+                id = "footnote-inpage_wikipedia";
+                title = _("In-page Wikipedia footnotes"),
+                description = _([[Show footnotes at the bottom of pages in Wikipedia EPUBs.]]),
+                css = [[
+ol.references > li {
+    -cr-hint: footnote-inpage;
     margin-top: 0 !important;
     margin-bottom: 0 !important;
 }
-/* Wikipedia EPUBs: hide backlinks */
+/* hide backlinks */
 ol.references > li > .noprint { display: none; }
-            ]],
+ol.references > li > .mw-cite-backlink { display: none; }
+                ]],
+            },
+            {
+                id = "footnote-inpage_wikipedia_smaller";
+                title = _("In-page Wikipedia footnotes (smaller)"),
+                description = _([[Show footnotes at the bottom of pages in Wikipedia EPUBs.]]),
+                css = [[
+ol.references > li {
+    -cr-hint: footnote-inpage;
+    margin-top: 0 !important;
+    margin-bottom: 0 !important;
+    font-size: 80% !important;
+}
+/* hide backlinks */
+ol.references > li > .noprint { display: none; }
+ol.references > li > .mw-cite-backlink { display: none; }
+                ]],
+                separator = true,
+            },
+            -- We can add other classic class names to the 2 following
+            -- tweaks (except when named 'calibreN', as the N number is
+            -- usually random across books).
+            {
+                id = "footnote-inpage_classic_classnames";
+                title = _("In-page classic footnotes"),
+                description = _([[
+Show footnotes with classic class names at the bottom of pages.
+This tweak can be duplicated as a user style tweak when books have footnotes wrapped with other class names.]]),
```

Comment:
Maybe something like this?
```suggestion
This tweak can be duplicated as a user style tweak when books contain footnotes indicated by other class names.]]),
```